sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala

```diff
@@ -824,8 +824,8 @@ class CsvFunctionsSuite extends QueryTest with SharedSparkSession {
       parameters = Map("schema" -> "\"STRUCT<a: VARIANT, b: VARIANT>\""))
 
     // In singleVariantColumn mode, from_csv normally treats all inputs as valid. The only exception
-    // case is the input exceeds the variant size limit (16MiB).
-    val largeInput = "a" * (16 * 1024 * 1024)
+    // case is the input exceeds the variant size limit (128MiB).
+    val largeInput = "a" * (128 * 1024 * 1024)
```

Comment:
Maybe the GA machines are not powerful enough to run tests that take 128 MB. One idea is to have a hardcoded testing limit of 16 MB. e.g.
```
public static final int SIZE_LIMIT = if (System.getenv("SPARK_TESTING") != null) U24_MAX + 1 else  128 * 1024 * 1024;
```

Reply:
then local test and GA test will have different error messages, we will have to complicate the test cases to make them pass for both environments.

Reply:
I've opened https://github.com/apache/spark/pull/50950